firestarter/workflows/build_images/auth.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+from abc import ABC, abstractmethod
+from typing import List
+import boto3.session
+import json
+from os import getenv, path
+
+def provider_from_str(provider: str) -> Provider:
+    if provider == 'aws_oidc':
+        return AWSProvider()
+    elif provider == 'azure_oidc':
+        return AzureProvider()
+    else:
+        raise ValueError(f'Unknown provider: {provider}')
+
+
+def login_registry(registry, auth):
+    docker_cfg_path = f"{getenv('HOME')}/.docker/config.json"
+    if not path.exists(docker_cfg_path):
+        config = {}
+    else:
+        with open(docker_cfg_path, 'r') as f:
+            config = json.load(f)
```

Comment:
I would put this methods in a factory class, as class methods,  something like :

`DockerRegistryAuthFactory.provider_from_str(provider=xxx) -> Provider`

Also I would put the `login_registry` method in the base class (no abstract class, so  no ABC heritage).
Also I would rename the Provider(s) with a more concrete name:
- DockerRegistryAuth
- AwsOidcDockerRegistryAuth
- AzureOidcDockerRegistryAuth
- GenericDockerRegistryAuth # by token